oci/layout/oci_transport.go

```diff
@@ -72,11 +73,11 @@ func ParseReference(reference string) (types.ImageReference, error) {
 	return NewReference(dir, image)
 }
```

Comment:
Because `StringWithinTransport` will (see elsewhere) need to contain the index, `ParseReference` in oci/layout also needs to support a string form of the index syntax (ideally sharing much of the code with oci/archive; and then docs/containers-transports.5.md will need to be updated).